mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -124,54 +134,61 @@ public String getMailetInfo() {
-        if (mail.getAttribute(sourceAttributeName) == null) {
-            return;
-        }
-        if (mail.getAttribute(rawSourceAttributeName) == null) {
+        if (!mail.getAttribute(AttributeName.of(sourceAttributeName)).isPresent()) {
             return;
         }
-        Optional<String> sender = retrieveSender(mail);
-        if (!sender.isPresent()) {
-            LOGGER.info("Skipping {} because no sender and no from", mail.getName());
+        if (!mail.getAttribute(AttributeName.of(rawSourceAttributeName)).isPresent()) {
             return;
         }
-        try {
-            Map<String, Calendar> calendars = getCalendarMap(mail);
-            Map<String, byte[]> rawCalendars = getRawCalendarMap(mail);
-            Map<String, byte[]> jsonsInByteForm = calendars.entrySet()
-                .stream()
-                .flatMap(calendar -> toJson(calendar, rawCalendars, mail, sender.get()))
-                .collect(Guavate.toImmutableMap(Pair::getKey, Pair::getValue));
-            mail.setAttribute(destinationAttributeName, (Serializable) jsonsInByteForm);
-        } catch (ClassCastException e) {
-            LOGGER.error("Received a mail with {} not being an ICAL object for mail {}", sourceAttributeName, mail.getName(), e);
-        }
+
+        retrieveSender(mail)
+            .map(sender -> {
+                try {
+                    getCalendarMap(mail)
+                        .ifPresent(calendars -> {
+                            getRawCalendarMap(mail)
+                                .ifPresent(rawCalendars -> {
+                                    Map<String, byte[]> jsonsInByteForm = calendars.entrySet()
+                                        .stream()
+                                        .flatMap(calendar -> toJson(calendar, rawCalendars, mail, sender))
+                                        .collect(Guavate.toImmutableMap(Pair::getKey, Pair::getValue));
+                                    mail.setAttribute(new Attribute(AttributeName.of(destinationAttributeName), AttributeValue.ofAny(jsonsInByteForm)));
+                                });
+                        });
+                } catch (ClassCastException e) {
+                    LOGGER.error("Received a mail with {} not being an ICAL object for mail {}", sourceAttributeName, mail.getName(), e);
+                }
+                return null;
+            }).orElseGet(() -> {
+                LOGGER.info("Skipping {} because no sender and no from", mail.getName());
+                return null;
+            });
     }
 
     @SuppressWarnings("unchecked")
-    private Map<String, Calendar> getCalendarMap(Mail mail) {
-        return (Map<String, Calendar>) mail.getAttribute(sourceAttributeName);
+    private Optional<Map<String, Calendar>> getCalendarMap(Mail mail) {
+        return AttributeUtils.getValueAndCastFromMail(mail, AttributeName.of(sourceAttributeName), (Class<Map<String, Calendar>>)(Object) Map.class);
```

Comment:
`(Class<Map<String, Calendar>>)(Object) Map.class` can be extracted as a constant